js/app/dashboard/redux/actions.js

```diff
@@ -1567,16 +1580,16 @@ export function deleteTour(tour) {
   }
 }
 
-export function removeTaskFromTour(tour, task, username) {
+export function removeTaskFromTour(tour, task, username, unassignTasksAction=unassignTasks, modifyTourAction=modifyTour) {
```

Comment:
If `unassignTasksAction=unassignTasks, modifyTourAction=modifyTour` are only needed for testing maybe we can find another way to test this function? For example, checking `expect` on the `dispatch` mock?

Reply:
in this particular case I don't remember why I added this as arg because it doesnt seem to be used (???) I will delete it

generally however I think I can only test with `expect` that dispatch has been called with a `function` as argument, but I would like to test the arguments that were passed to this `function`, so I think i need the `function`  to be a mock

is it clear?

Reply:
I see. The only other approach that I found is to move it into a separate file and mock the `unassignTasks` and `modifyTour` imports: https://jestjs.io/docs/mock-functions#mocking-modules

Reply:
yes  I started with that, I usually do mock modules, and yes if you move stuff to another file it works
but while working on this I realise that dependency injection is much better because it doesnt make you depend on the import logic/order which is a pain because you never  are sure you get the mocked version of the function in your test (like moving stuff or changing the import chain will break the mocking)

i realised that for this project and in general ahah